agents/loco_mc_agent.py

```diff
@@ -276,22 +269,36 @@ def controller_step(self):
             if chat.startswith("/"):
                 continue
             incoming_chats.append((speaker, chat))
-            self.memory.add_chat(self.memory.get_player_by_name(speaker).memid, chat)
 
         if len(incoming_chats) > 0:
             # force to get objects, speaker info
             if self.perceive_on_chat:
-                self.perceive(force=True)
-            # change this to memory.get_time() format?
+                force = True
             self.last_chat_time = time.time()
-            # to here ###########################################
-            # for now just process the first incoming chat
-            self.dialogue_manager.step(incoming_chats[0])
-        else:
+            # For now just process the first incoming chat, where chat -> [speaker, chat]
+            speaker, chat = incoming_chats[0]
+            preprocessed_chat, chat_parse = self.chat_parser.get_parse(chat)
+            # add postprocessed chat here
+            chat_memid = self.memory.add_chat(self.memory.get_player_by_name(speaker).memid, preprocessed_chat)
+            logical_form_memid = self.memory.add_logical_form(chat_parse)
```

Comment:
as mentioned in the sql_memory.py comment, I think its better to do a ProgramNode.create(), but I don't have strong opinion.  longer term, do you want to keep the main memory creation?  we should do one or the other; and if we standardize on main memory then I think we should have a more generic creation interface

Reply:
Yeah I  have a preference for keeping these as methods / APIs exposed from memory instead of directly accessing from memory nodes, where possible.